encoder/encoder.proto

```diff
@@ -75,14 +75,58 @@ message CreateStreamRequest {
 
   // An enumeration which allows us to express whether the device will be
   // located indoors or outdoors when deployed.
-  enum Disposition {
+  enum Exposure {
     INDOOR = 0;
     OUTDOOR = 1;
   }
 
-  // The specific disposition of the device, i.e. is this instance indoors or
+  // The specific exposure of the device, i.e. is this instance indoors or
   // outdoors.
-  Disposition disposition = 7;
+  Exposure exposure = 7;
+
+  message Configuration {
+    // The unique id of the sensor type for which this specific configuration is
+    // defined. This is a required field.
+    uint32 sensor_id = 1;
+
+    // An enumeration which allows us to specify what type of sharing is to be
+    // defined for the specified sensor type. The default value is `SHARE` which
+    // implies sharing the data at full resolution. If this type is specified, it
+    // is an error if either of `buckets` or `interval` is also supplied.
+    enum Action {
```

Comment:
An enum specifies an exclusive selection, maybe I'm no t understanding well but I think the actions should be inclusive, enabling to combine those "actions" , I think it shold be expressed maybe as bool flags?

Reply:
Probably "share" should be at least a separate option.

Reply:
can you explain what you mean? I'd been thinking about each individual "configuration" as exclusive so perhaps you have a different mental model. 

For me I'd been thinking about one logical "stream" as being comprised of one or more configurations which each refer to a specific sensor ID (i.e. a direct translation of the entitlement policy expressed on the device registration service).

So a stream might define binned sound levels (below 40 dBC, between 40 and 100 dBA, and above 100 dBA, as well as a 5 minute moving average light levels, and that would be expressed as a request of:

```json
{
  "brokerAddress": "tcp://foobar:1883",
  "deviceTopic": "/device/sck/absdhs/readings",
  "devicePrivateKey": "absdjd8u/d8guy8sd",
  "recipientPublicKey": "sjlasdkgaksjdglkajslbajdh=",
  "userUid": "alice",
  "location": {
    "longitude": -0.0025,
    "latitude": 54.2466
  },
  "exposure": "OUTDOOR",
  "configurations": [
    {
      "sensorId": 29,
      "action": "BIN",
      "bins": [40, 100]
    },
    {
      "sensorId": 13,
      "action": "MOVING_AVG",
      "interval": 300
    }
  ]
}
```

So for me each configuration is exclusive, I don't think they overlap; binning and averaging doesn't make sense.

I think there is an open question about whether we would allow a single sensorId to be used in multiple configurations, i.e. in the above should we allow a single stream to collect both a moving average and a binned value for noise level, or should that be an error.

Reply:
Yeah, got it, thanks!